### ISSUE TITLE
set resolution via tcp protocol

### ISSUE THREAD

**Issue by Licenser** [dalmatinerdb/dalmatinerdb#64]
A second version of the stream command that allows setting the bucket resolution would prevent us from having to pre-set this via the cli

**Comment by davecromberge** [comment]
Mind if I have a go at this one?

**Comment by Licenser** [comment]
go for it :)

**Comment by davecromberge** [comment]
In the context of setting TTL, Resolution etc as part of the stream, having them optional may introduce some complexity with regard to unset values, which could lead to too many variations in the resulting binary if we allow setting one without the other.  I would be tempted to include a suitable default if the value is not supplied.  The other possibility is to include a NO-OP binary value to describe that the value is unset.  Finally, when getting the bucket in the metric_vnode the pattern match will serve as validation that the values match.

For example, possible variations include:
```erlang
<<?STREAM,
      Delay:?DELAY_SIZE/?SIZE_TYPE,
      (byte_size(Bucket)):?BUCKET_SS/?SIZE_TYPE, Bucket/binary>>

<<?STREAM,
      Delay:?DELAY_SIZE/?SIZE_TYPE,
      Resolution:?TIME_SIZE/?TIME_TYPE,
      (byte_size(Bucket)):?BUCKET_SS/?SIZE_TYPE, Bucket/binary>>

And so on for TTL...

```

**Comment by Licenser** [comment]
TTL doesn't need to be set as part of the stream command. Unlike resolution it can be done after the fact and changed so I think it's sensible to have a own 'set ttl' value. For the two examples above those are distinct, one can never match the other.

**Comment by davecromberge** [comment]
Good point about TTL, I was just concerned about all the different shapes of the binary, even though they would be mutually exclusive - but that is no longer a concern seeing as TTL can be set separately 👍

**Comment by Licenser** [comment]
TODO:
 [ ] Adjust ddb itself
 [ ] update ddb cleint
